python/lsst/ctrl/bps/wms/panda/edgenode/cmd_line_decoder.py

```diff
@@ -4,12 +4,30 @@
 cluster management -> Edge node -> Container. Once the command line hexified it can contain symbols often
 stripped such as ',",` and multiline commands.
 """
+import os
+import re
 import sys
 import binascii
-cmdline = str(binascii.unhexlify(sys.argv[1]).decode())
-dataparams = sys.argv[2].split("+")
-cmdline = cmdline.replace("${filename}", dataparams[0])
-if len(dataparams) > 2:
-    cmdline = cmdline.replace("${qgraph-id}", dataparams[2])
-    cmdline = cmdline.replace("${qgraph-node-id}", dataparams[3])
-print(cmdline)
+
+
+def replace_environment_vars(cmdline):
+    envs_to_replace = re.findall(r'<ENV:(.*?)>', cmdline)
+    for env_var in envs_to_replace:
+        if os.getenv(env_var):
+            cmdline = cmdline.replace('<ENV:'+env_var+'>', os.getenv(env_var))
+    return cmdline
+
+
+cmd_line = str(binascii.unhexlify(sys.argv[1]).decode())
+data_params = sys.argv[2].split("+")
+cmd_line = replace_environment_vars(cmd_line)
+cmd_line = cmd_line.replace("<FILE:runQgraphFile>", data_params[0])
+qgraphId = data_params[2].split(":")[1]
+qgraphNodeId = data_params[3].split(":")[1]
+
+if len(qgraphId) > 0:
+    cmd_line = cmd_line.replace("{qgraphId}", qgraphId)
+
+if len(qgraphNodeId) > 0:
+    cmd_line = cmd_line.replace("{qgraphNodeId}", qgraphNodeId)
+print(cmd_line)
```

Comment:
If this is an actual executable python file we should document at the top that it's deliberately not importing any bps code for a reason. I guess we need to think about how plugins can publish their own additional commands and whether they can go in the path.

Reply:
you are right, everything what is under the _edgenode_ is the wrapper code executed on the edge node.